### PR TITLE
Add Soak Tests

### PR DESCRIPTION
**Pull request by NathanielRN**
*Description of changes:*

Similar to [the Soak Tests we have on Python](https://github.com/aws-observability/aws-otel-python), we now add Soak Tests to the ADOT Java Instrumentation repository.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

Check out [what changed in this PR from my fork](https://github.com/NathanielRN/aws-otel-java-instrumentation/pull/1).

### CODE REVIEW

File: .github/workflows/soak-testing.yml
Comment (anuraaga) [inline .github/workflows/soak-testing.yml:37]: https://github.com/NathanielRN/aws-otel-java-instrumentation/pull/1#discussion_r713714339

File: .github/workflows/soak-testing.yml
Comment (anuraaga) [inline .github/workflows/soak-testing.yml:74]: Yikes

Comment (NathanielRN) [reply .github/workflows/soak-testing.yml:74]: We discussed offline that the Soak Tests Design originally had a fundamental flaw. We should not be trying to publish snapshots during a Soak Test. We should just pull the image created during the Main build (or build a new image in the Soak Tests ourselves that is just like the main build).

The PR to fix this in the ADOT Python Downstream is here: https://github.com/aws-observability/aws-otel-python/pull/26

We will update this PR to match the result in that repo.

**Comment by anuraaga** [comment]
@NathanielRN Is this PR ready for review?

**Comment by NathanielRN** [comment]
@anuraaga No, it is waiting for approval and merging of https://github.com/NathanielRN/aws-otel-java-instrumentation/pull/1 first. I couldn't re-request your review because it's on my personal fork.

**Comment by NathanielRN** [comment]
Thanks for approving https://github.com/NathanielRN/aws-otel-java-instrumentation/pull/1 @anuraaga! With that, this PR is ready to merge. It simply takes the Soak Tests from AWS OTel Python and adds the changes needed for Java. Let me know if you think it’s ready to merge!

**Review by anuraaga — APPROVED** [review]

**Pull request merged**
